File: hc-mdm-web/src/main/java/com/hc/scm/mdm/web/controller/BasSupplierContacterController.java
package com.hc.scm.mdm.web.controller;

import com.hc.scm.common.base.service.BaseCrudService;
import com.hc.scm.common.base.web.BaseCrudController;
import com.hc.scm.mdm.dao.entity.BasSupplierContacter;
import com.hc.scm.mdm.service.BasSupplierContacterService;
import javax.annotation.Resource;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

/**
 * Description: 请写出类的用途
 * All rights Reserved, Designed Byhc* Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     Administrator
 * @date:  2015-03-31 13:51:25
 * @version 1.0.0
 */
@Controller
@RequestMapping("/bas_supplier_contacter")
public class BasSupplierContacterController extends BaseCrudController<BasSupplierContacter> {
    @Resource
    private BasSupplierContacterService basSupplierContacterService;

    @Override
    public BaseCrudService init() {
        return basSupplierContacterService;
    }
}

File: hc-mdm-web/src/main/webapp/resources/app/view/bassuppliercontacter/BasSupplierContacterController.js
/**
 * Description: 供应商联系人信息表Controller
 * All rights Reserved, Designed ByHc
 * Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     Administrator
 * @date:  2015-03-24 08:31:32
 * @version 1.0.0
 */
Ext.define('Hc_mdm.view.bassuppliercontacter.BasSupplierContacterController', {
    extend: 'Hc_Common.view.BaseSimplePageController',
    alias: 'controller.bassuppliercontacter',
    
    init: function (){
		var me = this;
		 me.callParent(arguments);
    },
    checkKeyValue:function(e){
		var me = this;
		 me.callParent(arguments);
		 var a=e;
		 var b=0;
    },
    onGridAfterEdit:function(editor, obj)
    {
		var me = this;
		 me.callParent(arguments);
    	if(obj.field == 'isDefault') {
			var record = obj.record;
			var isDefault = record.get('isDefault');
			var supplierNo = record.get('supplierNo');
			var contacterId = record.get('contacterId');
			if (isDefault==1){
            	Ext.Ajax.request({            
   		     		url: Hc.mdmPath+'bas_supplier_contacter/checkIsDefault.json',
   		     	    async:false,
   		     		params: {       
   		     			supplierNo:category_no
   		     		},    
   		     		success: function(response, options) {    
   		     			var result=response.responseText;
   		     			if (result!="0"){
   		     				Hc.alert(result);
   		     				//record.set('isDefault')
   		     			}
   		     		}
            	});
			}
    	}
    }
  });